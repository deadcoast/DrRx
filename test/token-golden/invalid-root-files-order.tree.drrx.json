[
  {
    "line": 1,
    "raw": ".                       # root",
    "tokens": [
      {
        "kind": "root",
        "start": 0,
        "end": 1,
        "text": "."
      },
      {
        "kind": "comment",
        "start": 24,
        "end": 30,
        "text": "# root"
      }
    ]
  },
  {
    "line": 2,
    "raw": "+-- src/                # directory before any root files (OR.01)",
    "tokens": [
      {
        "kind": "flow.open",
        "start": 0,
        "end": 1,
        "text": "+"
      },
      {
        "kind": "operator.dir",
        "start": 1,
        "end": 3,
        "text": "--"
      },
      {
        "kind": "operator.space",
        "start": 3,
        "end": 4,
        "text": " "
      },
      {
        "kind": "name.unquoted",
        "start": 4,
        "end": 8,
        "text": "src/"
      },
      {
        "kind": "name.gap",
        "start": 8,
        "end": 24,
        "text": "                "
      },
      {
        "kind": "comment",
        "start": 24,
        "end": 65,
        "text": "# directory before any root files (OR.01)"
      }
    ]
  },
  {
    "line": 3,
    "raw": "| :== main.py",
    "tokens": [
      {
        "kind": "flow.cont",
        "start": 0,
        "end": 1,
        "text": "|"
      },
      {
        "kind": "prefix.gap",
        "start": 1,
        "end": 2,
        "text": " "
      },
      {
        "kind": "flow.close",
        "start": 2,
        "end": 3,
        "text": ":"
      },
      {
        "kind": "operator.file",
        "start": 3,
        "end": 5,
        "text": "=="
      },
      {
        "kind": "operator.space",
        "start": 5,
        "end": 6,
        "text": " "
      },
      {
        "kind": "name.unquoted",
        "start": 6,
        "end": 13,
        "text": "main.py"
      }
    ]
  },
  {
    "line": 4,
    "raw": "+== README.md           # root file appears after directory (also OR.04)",
    "tokens": [
      {
        "kind": "flow.open",
        "start": 0,
        "end": 1,
        "text": "+"
      },
      {
        "kind": "operator.file",
        "start": 1,
        "end": 3,
        "text": "=="
      },
      {
        "kind": "operator.space",
        "start": 3,
        "end": 4,
        "text": " "
      },
      {
        "kind": "name.unquoted",
        "start": 4,
        "end": 13,
        "text": "README.md"
      },
      {
        "kind": "name.gap",
        "start": 13,
        "end": 24,
        "text": "           "
      },
      {
        "kind": "comment",
        "start": 24,
        "end": 72,
        "text": "# root file appears after directory (also OR.04)"
      }
    ]
  },
  {
    "line": 5,
    "raw": "",
    "tokens": []
  }
]