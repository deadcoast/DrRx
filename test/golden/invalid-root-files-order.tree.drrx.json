[
  { "rule": "OR.01", "line": 2, "column": 1 },
  { "rule": "OR.04", "line": 4, "column": 1 }
]
